tests/integration/test_orders_api.py

```diff
@@ -536,6 +536,37 @@ async def test_download_asset_md(tmpdir, session):
     assert Path(filename).name == 'metadata.json'
 
 
+@respx.mock
+@pytest.mark.asyncio
+async def test_checksum_success(tmpdir, order_description, oid, session):
+    # Mock an HTTP response for download
+    order_description['state'] = 'success'
```

Comment:
If you mock 2 responses (i.e., mock an order with 2 files), then you can test the functionality of your checksum function.

```
order_description['state'] = 'success'
dl_url1 = TEST_DOWNLOAD_URL + '/1?token=IAmAToken'
dl_url2 = TEST_DOWNLOAD_URL + '/2?token=IAmAnotherToken'
order_description['_links']['results'] = [{
    'location': dl_url1
}, {
    'location': dl_url2
}]

get_url = f'{TEST_ORDERS_URL}/{oid}'
mock_resp = httpx.Response(HTTPStatus.OK, json=order_description)
respx.get(get_url).return_value = mock_resp

mock_resp1 = httpx.Response(HTTPStatus.OK,
                            json={"name":"","files":[{"path":Path(tmpdir, "manifest.json").name,
                                                        "digests":{"md5":"anOriginMd5",
                                                                    "sha256":"anOriginSha256"}}]},
                            headers={
                                'Content-Type':
                                'application/json',
                                'Content-Disposition':
                                'attachment; filename="manifest.json"'
                            })
respx.get(dl_url1).return_value = mock_resp1

mock_resp2 = httpx.Response(HTTPStatus.OK,
                            json={"name":"","files":[{"path":Path(tmpdir, "manifest.json").name,
                                                        "digests":{"md5":"aReturnMd5",
                                                                    "sha256":"aReturnSha256"}}]},
                            headers={
                                'Content-Type':
                                'application/json',
                                'Content-Disposition':
                                'attachment; filename="someotherfile.json"'
                            })
respx.get(dl_url2).return_value = mock_resp2
```